### PR TITLE
feat: Add singleton logger

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
## Overview

Add a singleton logger that can be imported and used in all of the project. This allows to use the logger to be used independently of any object, we no longer need to pass the logger as a parameters in the functions. This software pattern for logging is the one used in the other PyAnsys libraries.

## Details

Currently, due to the autogenerated code, two loggers need to coexist. Ideally, we should only have a logger, or at least a root logger and children loggers depending on it. Anyway, there isn't any logging problem by having two loggers in the stdout.

### CODE REVIEW

**Review by RobPasMue — CHANGES_REQUESTED** [review]
Sounds like a nice enhancement, but you just implemented the class right? I am missing tests for the logger and some actual logs throughout the library. Please implement both things and request review again

File: src/ansys/meshing/prime/core/model.py [hunk 1]
```diff
@@ -368,4 +369,4 @@ def python_logger(self):
         >>> model.python_logger.setLevel(logging.DEBUG)
 
         """
-        return self._logger
+        return LOG
```
Comment (AlejandroFernandezLuces) [inline src/ansys/meshing/prime/core/model.py:372]: I did substitute the old logger with the new logger in all places where it was used with this line. I can write some tests for this.

Comment (RobPasMue) [reply src/ansys/meshing/prime/core/model.py:372]: Okay, got it.
Still, let's add some tests. Ensure you only have one class instance for example (i.e. check that the singleton is working fine)

**Review by RobPasMue — APPROVED** [review]
LGTM! Great job @AlejandroFernandezLuces

**Pull request merged**